scipy/stats/tests/test_fit.py

```diff
@@ -970,6 +970,20 @@ def test_params_effects(self):
         assert_equal(res3.fit_result.params.loc, -13.85)
         assert not np.allclose(res3.null_distribution, res1.null_distribution)
 
+    def test_custom_statistic(self):
+        # Test support for custom statistic function
+        def ks_plus(dist, data):
+            x = np.sort(data, axis=-1)
+            n = x.shape[-1]
+            cdfvals = dist.cdf(x)
+            return (np.arange(1.0, n + 1) / n - cdfvals).max(axis=-1)
+
+        rng = np.random.default_rng(9121950977643805391)
+        res = goodness_of_fit(stats.expon,
+                              stats.expon.rvs(size=100, random_state=rng),
+                              statistic=ks_plus,
+                              random_state=rng)
+        assert_allclose(res.statistic, 0.041086274)
```

Comment:
Please compute the reference value on the fly with `ks_1samp`, if that's where it's coming from. If needed, loosen the tolerance.

Reply:
Actually, the comment was just about the hard-coded reference value `0.041086274`. I realize now that `goodness_of_fit` is effectively performing a Lilliefors test here, so the p-value would not be super close to the one produced by `ks_1samp`. But still, where is that value coming from? I'd really prefer not to use the output of `goodness_of_fit` as a reference value. This is only appropriate when the intent is to "check that the output doesn't change", and that is rarely the best strategy available.

In any case, if the intent is really just to check that the statistic is called, maybe have the statistic raise a specific error and check for it (`with pytest.raises(..., match='your error message')`?

That would be OK, but I think it would be a stronger test to really show that the custom statistic is doing something useful. For instance, you could show that `goodness_of_fit` can perform something like [`skewtest`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.skewtest.html) and compare the output against that of `skewtest`. This would also make the test run faster because the normal distribution fits really fast - and you could even pass `known_params=dict(loc=0, scale=1)` because the location and scale don't affect the statistic.

Reply:
I get your point now. It all makes sense!

Let me rework on a more interesting test statistic than the one-sided KS. I have in mind some spacing-based goodness-of-fit tests. I'll try them out in a real-world problem next week and come back!

Reply:
I made a new commit that (1) calls the custom statistic with `axis=-1`, and (2) tests a different statistic against published numerical result. Hope it's good!

I leave the wording about `samples` unchanged because (1) I understand it is the correct usage, and (2) right below it the parameter `n_mc_samples` and its documentation use the word `samples` in the same way.

Reply:
I suppose if it's qualified as "Monte Carlo samples"  then it won't be confused.
Still "statistics" was used in two ways within the _same_ paragraph, so I adjusted that.
I also removed the `assert`s from production code and fixed an indention oops.

Reply:
Thank you very much for the detailed review!